### PR TITLE
Support break from general loop

### LINKED ISSUE #21: Doesn't support break in while loop
It is not valid now:

```
 while(1){
    break;
}
```

### PR DESCRIPTION
**Pull request by eecheng87**
Close #21

### CODE REVIEW

File: tests/driver.sh [hunk 1]
```diff
@@ -296,4 +296,16 @@ int main()
 }
 EOF
 
+# break from while
+try_ 1 << EOF
+int main()
+{
+    while(1) {
+        break;
+        abort();
```
Comment (jserv) [inline tests/driver.sh:305]: Add comment around `abort()`. For instance, "should not reach here."

File: tests/driver.sh [hunk 2]
```diff
@@ -296,4 +296,17 @@ int main()
 }
 EOF
 
+# break from while
```
Comment (jserv) [inline tests/driver.sh:299]: The existing section `loop` already contains `while(1) break; exit(10);`. You should enable it along with the similar cases.

File: tests/driver.sh [hunk 3]
```diff
@@ -296,4 +296,17 @@ int main()
...
+try_ 1 << EOF
```
Comment (jserv) [inline tests/driver.sh:300]: The following test program might be redundant if the test cases in `loop` section are enabled.

File: src/cfront.c [hunk 4]
```diff
@@ -1514,24 +1514,35 @@ void read_body_statement(block_t *parent)
 
     if (lex_accept(T_while)) {
         ir_instr_t *false_jump;
-        ir_instr_t *start = add_instr(OP_label); /* start to return to */
-
+        ir_instr_t *start_jump =
```
Comment (jserv) [inline src/cfront.c:1517]: Can you confirm the case `for (;;) break;` as well?

Comment (eecheng87) [reply src/cfront.c:1517]: should I send another PR? or also support `for (;;) break;` in this PR?

Comment (jserv) [reply src/cfront.c:1517]: Let's consolidate the "break" support for both `while` and `for` loop in this pull request.

Comment (eecheng87) [reply src/cfront.c:1517]: Now break is valid in `do_while`, `for` and `while` !

Comment (lazyparser) [reply src/cfront.c:1517]: good job

**Pull request merged**